lib/datadog/ai_guard/api_client.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require "uri"
+require "net/http"
+require "json"
+
+module Datadog
+  module AIGuard
+    # API Client for AI Guard API.
+    # Uses net/http to perform request. Raises on client and server errors.
+    class APIClient
+      DEFAULT_SITE = "app.datadoghq.com"
+
+      class UnexpectedRedirectError < StandardError; end
+
+      class UnexpectedResponseError < StandardError; end
+
+      class ResponseBodyParsingError < StandardError; end
+
+      class ClientError < StandardError; end
+
+      class NotFoundError < StandardError; end
+
+      class TooManyRequestsError < ClientError; end
+
+      class UnauthorizedError < ClientError; end
+
+      class ForbiddenError < ClientError; end
+
+      class ServerError < StandardError; end
+
+      class ReadTimeout < StandardError; end
+
+      def initialize(endpoint:, api_key:, application_key:, timeout:)
+        @api_key = api_key
+        @application_key = application_key
+        @timeout = timeout
+        @site = Datadog.configuration.site || DEFAULT_SITE
+        @endpoint = endpoint
+      end
+
+      def post(path:, request_body:)
```

Comment:
I think path could be a normal argument here, at the same time we can shorten `request_body` to be just body of the post request, WDYT?

```suggestion
      def post(path, body:)
```